### PR TITLE
Deprecate config item `max-memory` and add items `server-memory-quota` and `memory-usage-alarm-ratio` (#4333)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #4333 to release-4.0

---

### What is changed, added, or deleted? (Required)

Deprecate config item max-memory and add items server-memory-quota and memory-usage-alarm-ratio (only in dev)

max-memory has been deprecated and use server-memory-quota instead now. (only in dev)
memory-usage-alarm-ratio is introduced in this pr pingcap/tidb#18858.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4977
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@CharLotteiu Please resolve conflicts and apply version-specific changes according to https://github.com/pingcap/docs-cn/pull/5009.

**Comment by CharLotteiu** [comment]
@TomShawn PTAL again.

File: system-variables.md [hunk 1]
```diff
@@ -823,3 +823,41 @@ This variable is an alias for _transaction_isolation_.
...
+explain select * from t where age=5;
++-------------------------------+------------+-----------+-----------------------------+-------------------------------+
+| id                            | estRows    | task      | access object               | operator info                 |
++-------------------------------+------------+-----------+-----------------------------+-------------------------------+
+| IndexLookUp_7                 | 1048576.00 | root      |                             |                               |
+| ├─IndexRangeScan_5(Build)     | 1048576.00 | cop[tikv] | table:t, index:idx_age(age) | range:[5,5], keep order:false |
+| └─TableRowIDScan_6(Probe)     | 1048576.00 | cop[tikv] | table:t                     | keep order:false              |
++-------------------------------+------------+-----------+-----------------------------+-------------------------------+
+3 rows in set (0.00 sec)
+```
```
Comment (TomShawn) [inline system-variables.md:856]:
```suggestion
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Pull request merged**